### PR TITLE
[CDAP-4007] TestBase should not attempt to create default namespace

### PR DESCRIPTION
**Pull request by anew**
When multiple tests classes are run in sequence in the same JVM, then TestBase attempts to create the default namespace for every test class in the @BeforeClass method. This happens when selecting multiple tests in IDE and running them. It does not happen in maven (it uses a new JVM for every test) or inside a TestSuite.

The fix is to remember whether this is the first time the TestBase is initialized, and only create the default namespace in that case.

### CODE REVIEW

File: cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java [hunk 1]
```diff
@@ -291,9 +292,16 @@ protected void configure() {
       authorizerInstantiator.get().grant(NamespaceId.DEFAULT, principal, ImmutableSet.of(Action.ADMIN));
     }
     namespaceAdmin = injector.getInstance(NamespaceAdmin.class);
-    namespaceAdmin.create(NamespaceMeta.DEFAULT);
+    if (firstInit) {
```
Comment (albertshau) [inline cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java:295]: there is already a startCount which is checked at the start of this method. Does this do anything that check doesn't do? Actually I'm confused how this is still an issue with that check there.

Comment (albertshau) [inline cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java:295]: ok I see its decremented in the finish method

**Comment by albertshau** [comment]
lgtm

File: cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java (hunk 1 above)
Comment (anew) [inline cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java:295]: Yeah the startCount counts nested invocations, such as happens when run inside a TestSuite, the suite may extend from TestBase, and every Test will also extend from TestBase. Perhaps that should be renamed to nestedStartCount?

**Comment by anew** [comment]
Build running here http://builds.cask.co/browse/CDAP-RUT0

**Pull request merged**